usr/src/uts/common/fs/zfs/spa_history.c

```diff
@@ -455,6 +455,7 @@ log_internal(nvlist_t *nvl, const char *operation, spa_t *spa,
 	fnvlist_add_string(nvl, ZPOOL_HIST_INT_NAME, operation);
 	fnvlist_add_uint64(nvl, ZPOOL_HIST_TXG, tx->tx_txg);
 
+	zfs_ereport_spa_history(spa, nvl);
```

Comment:
This could generate a lot of events, and it's exposing a lot of internal details.  I'm not sure this is a great idea.